### PR TITLE
[DOCS] Relocate `shard allocation` module content

### PR DESCRIPTION
**Pull request by jrodewig**
Moves `Shard allocation and cluster-level` setting content from `Modules` to 
`Configuring Elasticsearch`.

This content was previously spread across multiple pages but is now
condensed to a single page. Adds several related redirects.

Relates to #53303.

### CODE REVIEW

File: docs/reference/modules/cluster/misc.asciidoc [hunk 1]
```diff
@@ -127,7 +126,7 @@ PUT /_cluster/settings
 
 
 [[persistent-tasks-allocation]]
-==== Persistent Tasks Allocations
+===== Persistent tasks allocations
```
Comment (debadair) [inline docs/reference/modules/cluster/misc.asciidoc:130]: The plural here seems odd--and "a kind of tasks" seems just wrong. (As does "long-live tasks".)

File: docs/reference/modules/cluster/shards_allocation.asciidoc [hunk 2]
```diff
@@ -1,15 +1,9 @@
...
-when nodes are added or removed.
-
-[float]
-=== Shard allocation settings
+[[cluster-shard-allocation-settings]]
+==== Cluster-level shard allocation settings
 
 The following _dynamic_ settings may be used to control shard allocation and recovery:
 
+[[cluster.routing.allocation.enable]]
```
Comment (debadair) [inline docs/reference/modules/cluster/shards_allocation.asciidoc:6]: This anchor should use dashes instead of dots.

**Review by debadair — APPROVED** [review]
The dots in the anchor are problematic...other than that LGTM.

File: docs/reference/modules/cluster/misc.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/modules/cluster/misc.asciidoc:130]: Thanks. Updated both.

File: docs/reference/modules/cluster/shards_allocation.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/modules/cluster/shards_allocation.asciidoc]: 👍

**Comment by jrodewig** [comment]
Thanks @debadair

**Comment by jrodewig** [comment]
### Backport commits
master 7c44931 
7.x 6934264 
7.8 84de077 
7.7 1e5ccf4

**Pull request merged**